src/Microsoft.IdentityModel.Tokens/Validation/LogDetail.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.IdentityModel.Abstractions;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Contains information so that logs can be written when needed.
+    /// </summary>
+    internal class LogDetail
```

Comment:
Did we consider LogDetail and MessageDetail as structs? Could avoid the allocations